source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -4720,17 +4731,15 @@ export class DateInput extends Input {
             this._dateInputElement.value = this.defaultValue;
         }
 
-        if (this.inputStyle !== null && this.inputStyle === Enums.InputStyle.ReadWrite) {
-            this._dateInputElement.classList.add(this.hostConfig.makeCssClassName("ac-dateInput-readWriteStyle"));
-            Utils.SetDateOrTimePickerVisibility(this._dateInputElement, this.defaultValue, false /*show*/);
+        return this._dateInputElement;
+    }
 
-            const handleHoverOnCard = (input: HTMLInputElement, eventType: Enums.CardEventType) => 
-                Utils.SetDateOrTimePickerVisibility(input, this._dateInputElement.value, eventType === Enums.CardEventType.OnMouseEnterOnCard /*show*/);
+    updateVisualState(hover: boolean): void {
+        super.updateVisualState(hover);
 
-            this.registerEventsForReadWriteInputStyle( this._dateInputElement, handleHoverOnCard.bind(this));
+        if (this._dateInputElement && this.inputStyle === Enums.InputStyle.RevealOnHover) {
+            Utils.updateDateOrTimePickerVisibility(this._dateInputElement, hover);
```

Comment:
This utilisty function takes an `HTMLInputElement` as a paremeter, so maybe naming it `updateInputVisibility` makes more sense?

Also, that function as part of `Utils` is exported and thus usable by developers that consume the SDK. However it doesn't make much sense to them as it's ultra specific to what you need to do. I suggest moving it to `card-elements.ts` as  non-exported function.